test/crypto/elliptic.js

```diff
@@ -460,31 +499,67 @@ describe('Elliptic Curve Cryptography', function () {
+      const ECDHE_Z2 = await genPrivateEphemeralKey("curve25519", ECDHE_VZ2.V, Q2, d2, fingerprint2);
       expect(Array.from(ECDHE_Z2).join(' ') === Array.from(ECDHE_VZ2.Z).join(' ')).to.be.false;
     });
     it('Different keys', async function () {
       const ECDHE_VZ1 = await genPublicEphemeralKey("curve25519", Q1, fingerprint1);
       const ECDHE_VZ2 = await genPublicEphemeralKey("curve25519", Q2, fingerprint1);
-      const ECDHE_Z1 = await genPrivateEphemeralKey("curve25519", ECDHE_VZ1.V, d1, fingerprint1);
+      const ECDHE_Z1 = await genPrivateEphemeralKey("curve25519", ECDHE_VZ1.V, Q1, d1, fingerprint1);
       expect(Array.from(ECDHE_Z1).join(' ') === Array.from(ECDHE_VZ2.Z).join(' ')).to.be.false;
     });
-    it('Successful exchange', async function () {
+    it('Successful exchange curve25519', async function () {
       const ECDHE_VZ1 = await genPublicEphemeralKey("curve25519", Q1, fingerprint1);
-      const ECDHE_Z1 = await genPrivateEphemeralKey("curve25519", ECDHE_VZ1.V, d1, fingerprint1);
+      const ECDHE_Z1 = await genPrivateEphemeralKey("curve25519", ECDHE_VZ1.V, Q1, d1, fingerprint1);
       expect(Array.from(ECDHE_Z1).join(' ') === Array.from(ECDHE_VZ1.Z).join(' ')).to.be.true;
     });
+    it('Successful exchange NIST P256', async function () {
+      const ECDHE_VZ1 = await genPublicEphemeralKey("p256", key_data.p256.pub, fingerprint1);
+      const ECDHE_Z1 = await genPrivateEphemeralKey("p256", ECDHE_VZ1.V, key_data.p256.pub, key_data.p256.priv, fingerprint1);
+      expect(Array.from(ECDHE_Z1).join(' ') === Array.from(ECDHE_VZ1.Z).join(' ')).to.be.true;
+    });
+    it('Successful exchange NIST P384', async function () {
+      const ECDHE_VZ1 = await genPublicEphemeralKey("p384", key_data.p384.pub, fingerprint1);
+      const ECDHE_Z1 = await genPrivateEphemeralKey("p384", ECDHE_VZ1.V, key_data.p384.pub, key_data.p384.priv, fingerprint1);
+      expect(Array.from(ECDHE_Z1).join(' ') === Array.from(ECDHE_VZ1.Z).join(' ')).to.be.true;
+    });
+    it('Successful exchange NIST P521', async function () {
+      const ECDHE_VZ1 = await genPublicEphemeralKey("p521", key_data.p521.pub, fingerprint1);
+      const ECDHE_Z1 = await genPrivateEphemeralKey("p521", ECDHE_VZ1.V, key_data.p521.pub, key_data.p521.priv, fingerprint1);
+      expect(Array.from(ECDHE_Z1).join(' ') === Array.from(ECDHE_VZ1.Z).join(' ')).to.be.true;
+    });
+    it('Comparing key derived using different algorithms', async function () {
+      const names = ["p256", "p384", "p521"];
+      if (openpgp.util.detectNode()) {
+          this.skip();
+      }
+      return Promise.all(names.map(async function (name) {
+        const curve = new elliptic_curves.Curve(name);
+        const key = window.crypto.subtle.generateKey({
+          name: "ECDSA",
+          namedCurve: curve.web.web
+        }, false, ["sign", "verify"]);
+        key.then(async () => {
```

Comment:
```suggestion
        return key.then(async () => {
```
Or switch to `await` - but this generated `key` seems to be unused? I guess the intention was to not run the test at all when it's not supported, you could do something like
```js
        try {
          await window.crypto.subtle.generateKey({
            name: "ECDSA",
            namedCurve: curve.web.web
          }, false, ["sign", "verify"]);
        } catch(err) {
          openpgp.util.print_debug_error(err);
          return;
        }
```
Currently, the checks below don't cause the test to fail even if key generation succeeded, and it seems they do fail for me in Chrome atm.